state/self_check.go

```diff
@@ -0,0 +1,262 @@
+package state
+
+import (
+	"fmt"
+	"sync"
+)
+
+// The self-check mechanism is intended to catch errors and potenttial problems
+// encountered during a test surface them to users in a test summary, and
+// communicate how these errors will impact the various features of pganalyze.
```

Comment:
For code review, start here. Most of the other code just calls this code with various messages.